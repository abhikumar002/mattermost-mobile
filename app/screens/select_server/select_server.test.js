// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
// See LICENSE.txt for license information.
import React from 'react';

import SelectServer from './select_server.js';
import {shallowWithIntl} from 'test/intl-test-helper';

describe('SelectServer', () => {
    const actions = {
        getPing: jest.fn(),
        handleServerUrlChanged: jest.fn(),
        scheduleExpiredNotification: jest.fn(),
        loadConfigAndLicense: jest.fn(),
        login: jest.fn(),
        resetPing: jest.fn(),
        setLastUpgradeCheck: jest.fn(),
        setServerVersion: jest.fn(),
    };

    const baseProps = {
        actions,
        hasConfigAndLicense: true,
        serverUrl: '',
    };

    test('should match snapshot empty URL string', async () => {
        const wrapper = shallowWithIntl(
            <SelectServer {...baseProps}/>,
        );
        wrapper.instance().handleConnect();
        expect(wrapper.state().error).not.toBeNull();
        expect(wrapper.state().error.intl.id).toMatch('mobile.server_url.empty');
        expect(wrapper.getElement()).toMatchSnapshot();
    });
});