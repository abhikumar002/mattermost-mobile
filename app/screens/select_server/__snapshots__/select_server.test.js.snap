// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SelectServer should match snapshot empty URL string 1`] = `
<ForwardRef(SafeAreaView)
  style={
    Object {
      "flex": 1,
    }
  }
  testID="select_server_screen"
>
  <KeyboardAvoidingView
    behavior="padding"
    enabled={true}
    keyboardVerticalOffset={0}
    style={
      Object {
        "flex": 1,
      }
    }
  >
    <StatusBar
      animated={false}
      barStyle="dark-content"
      showHideTransition="fade"
    />
    <TouchableWithoutFeedback
      accessible={false}
      onPress={[Function]}
    >
      <View
        style={
          Array [
            Object {
              "alignItems": "center",
              "backgroundColor": "white",
              "flex": 1,
              "flexDirection": "column",
              "justifyContent": "center",
            },
            Object {
              "paddingLeft": 15,
              "paddingRight": 15,
            },
          ]
        }
      >
        <Image
          source={
            Object {
              "testUri": "../../../dist/assets/images/logo.png",
            }
          }
        />
        <View>
          <FormattedText
            defaultMessage="Enter Server URL"
            id="mobile.components.select_server_view.enterServerUrl"
            style={
              Array [
                Object {
                  "fontSize": 32,
                  "fontWeight": "600",
                  "marginBottom": 15,
                  "marginTop": 15,
                  "textAlign": "center",
                },
                Object {
                  "fontSize": 20,
                  "fontWeight": "400",
                },
              ]
            }
          />
        </View>
        <Component
          allowFontScaling={true}
          autoCapitalize="none"
          autoCorrect={false}
          disableFullscreenUI={true}
          editable={true}
          keyboardType="url"
          onChangeText={[Function]}
          onSubmitEditing={[Function]}
          placeholder="https://mattermost.example.com"
          placeholderTextColor="rgba(0,0,0,0.5)"
          rejectResponderTermination={true}
          returnKeyType="go"
          style={
            Array [
              Object {
                "alignSelf": "stretch",
                "borderColor": "gainsboro",
                "borderRadius": 3,
                "borderWidth": 1,
                "color": "#3d3c40",
                "fontSize": 16,
                "height": 45,
                "marginBottom": 5,
                "marginTop": 5,
                "paddingLeft": 10,
              },
            ]
          }
          testID="server_url_input"
          underlineColorAndroid="transparent"
        />
        <Button
          containerStyle={
            Array [
              Object {
                "alignItems": "center",
                "alignSelf": "stretch",
                "borderColor": "#2389D7",
                "borderRadius": 3,
                "borderWidth": 1,
                "marginTop": 10,
                "padding": 15,
              },
              Object {
                "alignItems": "center",
              },
            ]
          }
          onPress={[Function]}
          testID="connect_button"
        >
          <Text
            style={
              Object {
                "color": "#2389D7",
                "fontSize": 17,
                "textAlign": "center",
              }
            }
          >
            <FormattedText
              defaultMessage="Connect"
              id="mobile.components.select_server_view.connect"
            />
          </Text>
        </Button>
        <View>
          <Connect(ErrorText)
            error={
              Object {
                "intl": Object {
                  "defaultMessage": "Please enter a valid server URL",
                  "id": "mobile.server_url.empty",
                },
              }
            }
          />
        </View>
      </View>
    </TouchableWithoutFeedback>
  </KeyboardAvoidingView>
</ForwardRef(SafeAreaView)>
`;
